src/mac/streams/thread_list.rs

```diff
@@ -0,0 +1,231 @@
+
+        // If the user tag is VM_MEMORY_STACK, look for more readable regions with
+        // the same tag placed immediately above the computed stack region. Under
+        // some circumstances, the stack for thread 0 winds up broken up into
+        // multiple distinct abutting regions. This can happen for several reasons,
+        // including user code that calls setrlimit(RLIMIT_STACK, ...) or changes
+        // the access on stack pages by calling mprotect.
+        if region.info.user_tag == mach2::vm_statistics::VM_MEMORY_STACK {
+            loop {
+                let proposed_next_region_base = region.range.end;
+
+                region = if let Ok(reg) = dumper.get_vm_region(region.range.end) {
+                    reg
+                } else {
+                    break;
+                };
+
+                if region.range.start != proposed_next_region_base
+                    || region.info.user_tag != mach2::vm_statistics::VM_MEMORY_STACK
+                    || (region.info.protection & mach2::vm_prot::VM_PROT_READ) == 0
+                {
+                    break;
+                }
+
+                stack_size += region.range.end - region.range.start;
+            }
+        }
+
+        (root_range_start + stack_size - start_address) as usize
+    }
+
+    pub(crate) fn fill_cpu_context(
+        thread_state: &crate::mac::mach::ThreadState,
+        out: &mut RawContextCPU,
+    ) {
+        let ts = thread_state.arch_state();
+
+        cfg_if::cfg_if! {
+            if #[cfg(target_arch = "x86_64")] {
+                out.context_flags = format::ContextFlagsCpu::CONTEXT_AMD64.bits();
+
+                out.rax = ts.__rax;
+                out.rbx = ts.__rbx;
+                out.rcx = ts.__rcx;
+                out.rdx = ts.__rdx;
+                out.rdi = ts.__rdi;
+                out.rsi = ts.__rsi;
+                out.rbp = ts.__rbp;
+                out.rsp = ts.__rsp;
+                out.r8 = ts.__r8;
+                out.r9 = ts.__r9;
+                out.r10 = ts.__r10;
+                out.r11 = ts.__r11;
+                out.r12 = ts.__r12;
+                out.r13 = ts.__r13;
+                out.r14 = ts.__r14;
+                out.r15 = ts.__r15;
+                out.rip = ts.__rip;
+                // according to AMD's software developer guide, bits above 18 are
+                // not used in the flags register.  Since the minidump format
+                // specifies 32 bits for the flags register, we can truncate safely
+                // with no loss.
+                out.eflags = ts.__rflags as _;
+                out.cs = ts.__cs as u16;
+                out.fs = ts.__fs as u16;
+                out.gs = ts.__gs as u16;
+            } else if #[cfg(target_arch = "aarch64")] {
+                // This is kind of a lie as we don't actually include the full float state..?
+                out.context_flags = format::ContextFlagsArm64Old::CONTEXT_ARM64_OLD_FULL.bits() as u64;
+
```

Comment:
The MacOS threadstate doesn't include floating point registers on aarch64, but Breakpad was still setting `MD_CONTEXT_ARM64_FULL_OLD` which includes `MD_CONTEXT_ARM64_FLOATING_POINT_OLD`. From what I have seen this is fine and stackwalker/sentry have no issues with the thread state in such a dump.